test/sanitization_test.js

```diff
@@ -1465,5 +1465,22 @@ exports.sanitization = function () {
 			candidate.should.be.eql(candidate);
 		});
 
+		test('candidate #4 | remove useless keys on custom classes', function () {
+			function G(obj) {
+				Object.keys(obj).forEach(key => {
+					this[key] = obj[key];
+				});
+			}
+
+			var candidate = new G({
+				good: 'key',
+				bad: 'key'
+			});
+
+			var result = si.sanitize(schema, candidate);
+			result.should.be.an.Object;
+			candidate.should.be.eql(new G({ good: 'key' }));
+		})
+
```

Comment:
Is this the behavior the issue raised in 2017 was asking for. it said "sanitize.strict" doesn't work for custom classes. Was it talking about needing the unwanted properties to be removed?

If so, can you add another test that uses a class defintion instead of a "constructor function"? I understand in JS they're basically the same, but I'd like to have a test too that reflects the kind of code people tend to write these days, which includes class definitions. Basically, just need another test that uses `class G...`.

Reply:
> Is this the behavior the issue raised in 2017 was asking for? It said "sanitize.strict" doesn't work for custom classes. Was it talking about needing the unwanted properties to be removed? (I was confused by the issue exactly what the ask was)

I am assuming so, because their schema only contained `contactId` while the results included `contactId` and `foo`.

> If so, can you add another test that uses a class defintion instead of a "constructor function"? I understand in JS they're basically the same, but I'd like to have a test too that reflects the kind of code people tend to write these days, which includes class definitions. Basically, just need another test that uses `class G....`

Sure

Reply:
893c772